tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,158 @@
+    import android._ctypes_library_finder   # actual import
+    sys.path.remove(temp_d)
+finally:
+    shutil.rmtree(temp_d)
+
+
+# Fake jnius module with autoclass function:
+import sys
+import types
+module_name = 'jnius'
+bogus_module = types.ModuleType(module_name)
+sys.modules[module_name] = bogus_module
+bogus_module.autoclass = lambda name: None
+
+
+def _get_autoclass_with_fake_activity(returning_test_path="/testpath",
+                                      return_as_service=False):
+    """
+    Helper functions to mock autoclass() from jnius.
+
+    This will return a mocked activity "org.kivy.android.PythonActivity"
+    which supports .getApplicationContext().getPackagemanager(),
+    with that returned object supporting .getApplicationInfo()
+    for PackageManager's GET_SHARED_LIBRARY_FILES info.
+
+    (That is the functionality the android._ctypes_library_finder module
+     uses to scan the application-specific library directories)
+    """
+    try:
+        from unittest.mock import MagicMock
+    except ImportError:  # Python 2.
+        from mock import MagicMock
+
+    def return_fake_app_context():  # Mocks activity.getApplicationContext()
+        context = MagicMock()
+        context.getPackageName = lambda: "com.example.testpackage"
+
+        def return_package_manager():  # Mocks context's .getPackageManager()
+            appinfo_obj = MagicMock()
+            appinfo_obj.nativeLibraryDir = returning_test_path
+            manager = MagicMock()
+            manager.getApplicationInfo = lambda x, y: appinfo_obj
+            return manager
+
+        context.getPackageManager = return_package_manager
+        return context
+
+    def autoclass_returning_graphical_activity(name):
+        if name == "org.kivy.android.PythonActivity" and \
+                not return_as_service:
+            activity = MagicMock()
+            activity.getApplicationContext = return_fake_app_context
+            activity_class = MagicMock()
+            activity_class.mService = None
+            activity_class.mActivity = activity
+            return activity_class
+        elif name == "org.kivy.android.PythonService" and return_as_service:
+            activity = MagicMock()
+            activity.getApplicationContext = return_fake_app_context
+            activity_class = MagicMock()
+            activity_class.mActivity = None
+            activity_class.mService = activity
+            return activity_class
+        elif name == "android.content.pm.PackageManager":
+            manager_class = MagicMock()
+            manager_class.GET_SHARED_LIBRARY_FILES = 1024
+            return manager_class
+        return None
+
+    return autoclass_returning_graphical_activity
+
+
+def test_get_activity_lib_dir():
+    import jnius  # should get us our fake module
+    jnius.autoclass = lambda name: None
+    assert(android._ctypes_library_finder.get_activity_lib_dir(
+        "org.kivy.android.PythonActivity"
+    ) is None)
```

Comment:
Please do not use parenthesis on your `assert` it's a keyword not a function

Reply:
I swear I don't do it on purpose, I'm just used to writing it like that so I keep forgetting :joy:

Reply:
changed it now :+1: thanks for the poke, will resolve for now